server/Controllers/WaterLevelSettingsController.cs

```diff
@@ -0,0 +1,94 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using WAMServer.Interfaces;
+using WAMServer.Models;
+using WAMServer.Records.Bodies;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WAMServer.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class WaterLevelSettingsController : ControllerBase
+    {
+        private readonly IRepository<WaterLevelSettings> _waterlevelsettingsRepository;
+        private readonly IWaterLevelSettings _WaterLevelSettingsService;
+        private readonly IRepository<User> _userRepository;
+
+        public WaterLevelSettingsController(IRepository<WaterLevelSettings> waterlevelsettingsRepository, IRepository<User> userRepository, IWaterLevelSettings waterLevelSettingsService)
+        {
+            _waterlevelsettingsRepository = waterlevelsettingsRepository;
+            _userRepository = userRepository;
+            _WaterLevelSettingsService = waterLevelSettingsService;
+        }
+        
+
+        [HttpPut]
+        [Authorize]
+        public async Task<IActionResult> Put([FromBody] WaterLevelSettingsPatchBody waterlevelsettingsPatchBody)
+        {
+            Console.WriteLine(waterlevelsettingsPatchBody);
+
+            const string unauthorizedmsg = "Errors.unauth";
+
+            if (waterlevelsettingsPatchBody == null || (waterlevelsettingsPatchBody.PoleHeight == "" && waterlevelsettingsPatchBody.IdealHeight == ""))
+            {
+                return BadRequest(new ErrorBody("Missing fields, incident reported."));
+            }
+
+            WaterLevelSettingsPatchBodyDecimal checkedSettings;
+
+
+            if (decimal.TryParse(waterlevelsettingsPatchBody.PoleHeight, out decimal poleDecimal) && decimal.TryParse(waterlevelsettingsPatchBody.IdealHeight, out decimal idealDecimal))
+            {
+                checkedSettings = new WaterLevelSettingsPatchBodyDecimal
+                {
+                    PoleHeight = poleDecimal,
+                    IdealHeight = idealDecimal
+                };            
+            }
+            else
+            {
+                return NotFound(new ErrorBody("Decimals not not found."));
```

Comment:
Deze error is niet zo duidelijk... Echter hoeft de gebruiker ook niet zo veel details... Mby iets van: "Kan op het moment niet..."

Reply:
dit is een error die alleen te zien is in de console, dus niet voor de gebruiker, ik dacht is voor debuggen duidelijker om specifiek te zijn